### PR TITLE
Add DOF = 0 assertion to initialization methods in 0D and 1DRO

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:

Address the last item on issue #91 task list: assert that the degrees of freedom are 0 within initialization method for RO model. 

## Changes proposed in this PR:
- add assert dof == 0 to initialize() in 0d and 1d RO

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by TimBartholomew** [comment]
It looks like tests are failing for the flowsheet, which is interesting. I wonder if I forgot to fix the width, or that I set a value but didn't fix the variable.

**Comment by adam-a-a** [comment]
> It looks like tests are failing for the flowsheet, which is interesting. I wonder if I forgot to fix the width, or that I set a value but didn't fix the variable.

@TimBartholomew It appears that there are 4 degrees of freedom when the initialize method is called in the RO flowsheet. Should the following lines be fixed instead in the test file? 

```
    m.fs.RO.feed_side.properties_in[0].flow_mass_phase_comp['Liq', 'H2O'] = \
        value(m.fs.feed.properties[0].flow_mass_phase_comp['Liq', 'H2O'])
    m.fs.RO.feed_side.properties_in[0].flow_mass_phase_comp['Liq', 'NaCl'] = \
        value(m.fs.feed.properties[0].flow_mass_phase_comp['Liq', 'NaCl'])
    m.fs.RO.feed_side.properties_in[0].temperature = \
        value(m.fs.feed.properties[0].temperature)
    m.fs.RO.feed_side.properties_in[0].pressure = \
        value(m.fs.P1.control_volume.properties_out[0].pressure)
```

**Comment by TimBartholomew** [comment]
>     m.fs.RO.feed_side.properties_in[0].flow_mass_phase_comp['Liq', 'H2O'] = \
>         value(m.fs.feed.properties[0].flow_mass_phase_comp['Liq', 'H2O'])
>     m.fs.RO.feed_side.properties_in[0].flow_mass_phase_comp['Liq', 'NaCl'] = \
>         value(m.fs.feed.properties[0].flow_mass_phase_comp['Liq', 'NaCl'])
>     m.fs.RO.feed_side.properties_in[0].temperature = \
>         value(m.fs.feed.properties[0].temperature)
>     m.fs.RO.feed_side.properties_in[0].pressure = \
>         value(m.fs.P1.control_volume.properties_out[0].pressure)

Oh! The degrees of freedom check should happen after the initialization fixes the state variables with unit.feed_side.properties_in.intialize(...,**hold_state=True**). 

Move the degrees of freedom check to just before the solve call.
```
# Solve unit
with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
   res = opt.solve(blk, tee=slc.tee)
```

**Review by bknueven — APPROVED** [review]

**Review by TimBartholomew — APPROVED** [review]
LGTM

File: proteuslib/unit_models/reverse_osmosis_1D.py [hunk 1]
```diff
@@ -585,7 +587,10 @@ def initialize(blk,
             state_args=permeate_block_args)
 
         init_log.info_high("Initialization Step 1 Complete.")
-
+        if degrees_of_freedom(blk) != 0:
+            raise Exception(f"Initialization was called on {blk} "
+                            f"but it had {degrees_of_freedom(blk)} degrees of freedom "
```
Comment (bknueven) [inline proteuslib/unit_models/reverse_osmosis_1D.py:592]:
```suggestion
                            f"but it had {degrees_of_freedom(blk)} degree(s) of freedom "
```
Might as well get this one too

**Review by bknueven — APPROVED** [review]

**Review by TimBartholomew — APPROVED** [review]

**Pull request merged**